### PR TITLE
Add default values for optional parameters.

### PR DESCRIPTION
**Pull request by scheglov**
Fixes violations when optional parameters specify different default values than the corresponding optional parameters in base classes. Dart SDK issue https://github.com/dart-lang/sdk/issues/34437.

For example, this `a` in `B.foo` is an error, because its default value is `2`, and the default value of `a` in `A.foo` is `1`.

```dart
class A {
  void foo({int a: 1}) {}
}

class B extends A {
  void foo({int a: 2}) {}
}
```

### CODE REVIEW

File: packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart [hunk 1]
```diff
@@ -185,7 +185,7 @@ class FuchsiaDevice extends Device {
     Map<String, dynamic> platformArgs,
     bool prebuiltApplication = false,
     bool applicationNeedsRebuild = false,
-    bool usesTerminalUi = false,
+    bool usesTerminalUi = true,
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/fuchsia/fuchsia_device.dart]: /cc @jonahwilliams

File: packages/flutter_tools/lib/src/commands/daemon.dart [hunk 2]
```diff
@@ -784,7 +784,7 @@ class NotifyingLogger extends Logger {
     String message, {
     @required Duration timeout,
     String progressId,
-    bool multilineOutput,
+    bool multilineOutput = false,
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/commands/daemon.dart]: /cc @devoncarew

File: packages/flutter_tools/lib/src/build_runner/build_kernel_compiler.dart [hunk 3]
```diff
@@ -33,7 +33,7 @@ class BuildKernelCompiler implements KernelCompiler {
     List<String> fileSystemRoots,
     String fileSystemScheme,
     String depFilePath,
-    TargetModel targetModel,
+    TargetModel targetModel = TargetModel.flutter,
```
Comment (tvolkert) [inline packages/flutter_tools/lib/src/build_runner/build_kernel_compiler.dart]: /cc @jonahwilliams

File: packages/flutter/lib/src/rendering/object.dart [hunk 4]
```diff
@@ -2633,7 +2633,7 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
   /// [toStringDeep], but does not recurse to any children.
   @override
   String toStringShallow({
-    String joiner = '; ',
+    String joiner = ', ',
```
Comment (tvolkert) [inline packages/flutter/lib/src/rendering/object.dart]: /cc @Hixie @goderbauer

File: packages/flutter_tools/lib/src/commands/daemon.dart (hunk 2 above)
Comment (devoncarew) [reply packages/flutter_tools/lib/src/commands/daemon.dart]: 👍

File: packages/flutter_tools/lib/src/build_runner/build_kernel_compiler.dart (hunk 3 above)
Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/build_runner/build_kernel_compiler.dart]: LGTM

**Comment by jonahwilliams** [comment]
All tool issues I was CC'd on: LGTM

**Comment by goderbauer** [comment]
Can you add a little explanation to the PR describing the motivation behind this change?

**Review by devoncarew — APPROVED** [review]
lgtm on a green build

cc @tvolkert

**Review by tvolkert — APPROVED** [review]

File: packages/flutter_test/lib/src/binding.dart [hunk 5]
```diff
@@ -720,7 +720,7 @@ class AutomatedTestWidgetsFlutterBinding extends TestWidgetsFlutterBinding {
 
   @override
   Future<T> runAsync<T>(Future<T> callback(), {
-    Duration additionalTime = const Duration(milliseconds: 1000),
+    Duration additionalTime = const Duration(milliseconds: 250),
```
[the remark this answers is not in the document]
Comment (tvolkert) [reply packages/flutter_test/lib/src/binding.dart]: Looks like this didn't get updated before merge.

[the remark this answers is not in the document]
Comment (scheglov) [reply packages/flutter_test/lib/src/binding.dart]: I'm sorry, I did not recognize that this is something that should be updated.
Do you want me to revert this PR?
Or send another one that updates both base class, and this to `1000`?

[the remark this answers is not in the document]
Comment (tvolkert) [reply packages/flutter_test/lib/src/binding.dart]: I think we should update both base class and this to be 1000 - follow-up PR is fine.

**Pull request merged**